src-plugins/medMaskApplication/medMaskApplication.cpp

```diff
@@ -33,56 +33,56 @@
 
 class medMaskApplicationPrivate
 {
-    public:
-        dtkSmartPointer <medAbstractData> mask;
-        dtkSmartPointer <medAbstractData> input;
-        dtkSmartPointer <medAbstractData> output;
-        double maskBackgroundValue;
+public:
+    dtkSmartPointer <medAbstractData> mask;
+    dtkSmartPointer <medAbstractData> input;
+    dtkSmartPointer <medAbstractData> output;
+    double maskBackgroundValue;
 
     template <class ImagePixelType, class MaskPixelType> int update()
     {
-        typedef itk::Image<ImagePixelType, 3> ImageType;
-        typedef itk::Image<MaskPixelType,  3> MaskType;
-
-        if ( !input ||!input->data() || !mask ||!mask->data())
-            return EXIT_FAILURE;
-
-        typedef itk::MaskImageFilter< ImageType,  MaskType> MaskFilterType;
-        typename MaskFilterType::Pointer maskFilter = MaskFilterType::New();
-
-        typename ImageType::Pointer imgInput = dynamic_cast<ImageType *> ( ( itk::Object* ) ( input->data() )) ;
-        typename MaskType::Pointer maskInput = dynamic_cast<MaskType *>  ( ( itk::Object* ) ( mask->data() )) ;
-
-        try
+        if (input->data() && mask->data())
         {
-            maskInput->SetOrigin(imgInput->GetOrigin());
-            maskInput->SetSpacing(imgInput->GetSpacing());
-            maskFilter->SetInput(imgInput);
-            maskFilter->SetMaskingValue(maskBackgroundValue);
-            maskFilter->SetMaskImage(maskInput);
-
-            //Outside values set to the lowest reachable value
-
-            typedef itk::MinimumMaximumImageCalculator <ImageType> ImageCalculatorFilterType;
- 
-            typename ImageCalculatorFilterType::Pointer imageCalculatorFilter
-                    = ImageCalculatorFilterType::New ();
-            imageCalculatorFilter->SetImage(imgInput);
-            imageCalculatorFilter->ComputeMinimum();
-            maskFilter->SetOutsideValue(std::min(double(imageCalculatorFilter->GetMinimum()), 0.0));
-            maskFilter->Update();
-            output->setData(maskFilter->GetOutput());
+            typedef itk::Image<ImagePixelType, 3 > ImageType;
+            typedef itk::Image<MaskPixelType,  3 > MaskType;
+
+            typedef itk::MaskImageFilter< ImageType,  MaskType> MaskFilterType;
+            typename MaskFilterType::Pointer maskFilter = MaskFilterType::New();
+
+            typename ImageType::Pointer imgInput = dynamic_cast<ImageType *> ( ( itk::Object* ) ( input->data() )) ;
+            typename MaskType::Pointer maskInput = dynamic_cast<MaskType *>  ( ( itk::Object* ) ( mask->data() )) ;
+
+            try
```

Comment:
Just a FYI,
the `try` could have been shortened, starting from `imageCalculatorFilter->ComputeMinimum();` and finishing with `maskFilter->Update();`